www/src/models/transformation/TransformationStore.ts

```diff
@@ -9,26 +9,47 @@ export interface Transformation extends Record<number> {
   tableId: number;
   datapath: string[];
   code: string | undefined;
+  onError: string;
   outputpath: string[] | undefined;
+}
+
+export interface DraftCreateTransformation extends Omit<Transformation, "id"> {
+  draftID: string;
   tolerance: number;
   rows: number;
 }
 
-export class TransformationStore extends SimpleCRUDStore<
+export interface DraftUpdateTransformation extends Transformation {
+  markSaved(): void;
+  toModel(): Transformation | undefined;
+}
+
+export class TransformationStore extends CRUDStore<
   number,
+  DraftCreateTransformation,
+  DraftUpdateTransformation,
   Transformation
 > {
   constructor() {
     super(`${SERVER}/api/transformation`, undefined, false);
   }
 
   async testTransformation(
-    payload: Transformation
+    payload: DraftCreateTransformation
   ): Promise<TransformationResult[] | undefined> {
     let resp: any = await axios
       .post(
-        `${SERVER}/api/transform/${payload.tableId}/transformations`,
-        payload
+        `${SERVER}/api/transformation/test?tolerance=${
+          payload.tolerance
+        }&rows=${payload!.rows}`,
+        {
+          table_id: payload.tableId,
+          type: payload.type,
+          code: payload.code,
+          mode: payload.mode,
+          datapath: payload.datapath,
+          outputpath: payload.outputpath,
+        }
       )
       .then((res) => res.data)
       .catch((error) => error.response.data.message);
```

Comment:
only catch and convert to non-error for 400 status code.

Reply:
updated and fixed